nominal/core/client.py

```diff
@@ -0,0 +1,497 @@
+            description=description or "",
+            labels=list(labels),
+            properties={} if properties is None else dict(properties),
+            s3_path=s3_path,
+            title=name,
+        )
+        response = self._clients.attachment.create(self._clients.auth_header, request)
+        return Attachment._from_conjure(self._clients, response)
+
+    def get_attachment(self, rid: str) -> Attachment:
+        """Retrieve an attachment by its RID."""
+        response = self._clients.attachment.get(self._clients.auth_header, rid)
+        return Attachment._from_conjure(self._clients, response)
+
+    def get_attachments(self, rids: Iterable[str]) -> Sequence[Attachment]:
+        """Retrive attachments by their RIDs."""
+        return list(_iter_get_attachments(self._clients, rids))
+
+    def get_all_units(self) -> Sequence[scout_units_api.Unit]:
+        """Retrieve list of all allowable units"""
+        response = self._clients.units.get_all_units(self._clients.auth_header)
+        return [unit for units in response.units_by_property.values() for unit in units]
+
+    def get_unit(self, unit_symbol: str) -> scout_units_api.Unit | None:
+        """Get details of the given unit symbol, or none if invalid"""
+        return self._clients.units.get_unit(self._clients.auth_header, unit_symbol)
+
+    def get_commensurable_units(self, unit_symbol: str) -> Sequence[scout_units_api.Unit]:
+        """Get the list of units that are commensurable (convertible to/from) the given unit symbol"""
+        return self._clients.units.get_commensurable_units(self._clients.auth_header, unit_symbol)
+
+
+def _create_search_runs_query(
+    start: datetime | IntegralNanosecondsUTC | None = None,
+    end: datetime | IntegralNanosecondsUTC | None = None,
+    name_substring: str | None = None,
+    label: str | None = None,
+    property: tuple[str, str] | None = None,
+) -> scout_run_api.SearchQuery:
+    queries = []
+    if start is not None:
+        q = scout_run_api.SearchQuery(start_time_inclusive=_SecondsNanos.from_flexible(start).to_scout_run_api())
+        queries.append(q)
+    if end is not None:
+        q = scout_run_api.SearchQuery(end_time_inclusive=_SecondsNanos.from_flexible(end).to_scout_run_api())
+        queries.append(q)
+    if name_substring is not None:
+        q = scout_run_api.SearchQuery(exact_match=name_substring)
+        queries.append(q)
+    if label is not None:
+        q = scout_run_api.SearchQuery(label=label)
+        queries.append(q)
+    if property is not None:
+        name, value = property
+        q = scout_run_api.SearchQuery(property=scout_run_api.Property(name=name, value=value))
+        queries.append(q)
+    return scout_run_api.SearchQuery(and_=queries)
+
+
+def _verify_csv_path(path: Path | str) -> tuple[Path, FileType]:
+    path = Path(path)
+    file_type = FileType.from_path_dataset(path)
+    if file_type.extension not in (".csv", ".csv.gz"):
```

Comment:
Should this super me a class constant in FileType? (Assuming we wrote that)... some ALLOWED_EXTENSIONS?

Reply:
Hmm, there is a diff between what's allowed by the API vs what's allowed by certain functions. In this case the function is validating that the path is an allowed CSV explicitly